@here/harp-materials/lib/RawShaderMaterial.ts

```diff
@@ -50,5 +52,12 @@ export class RawShaderMaterial extends THREE.RawShaderMaterial {
             delete (shaderParams as any).rendererCapabilities;
         }
         super(shaderParams);
+        this.invalidateFog();
+    }
+
+    protected invalidateFog() {
+        if (this.defines !== undefined && this.fog !== getShaderMaterialDefine(this, "USE_FOG")) {
+            setShaderMaterialDefine(this, "USE_FOG", this.fog);
+        }
```

Comment:
The problem now is that without the setter in SolidLineMaterial, the material defines are not updated when fog is enabled/disabled, so the shader won't change even if it's recompiled.
Since the fog settings are controlled only from MapViewFog, we can make this method invalidateFog() public and call it in MapViewFog.setFogInRawShaderMaterials().